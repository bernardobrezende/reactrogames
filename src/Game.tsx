import PropTypes, { InferProps } from 'prop-types';

export default function Game({
    title,
    platform,
    releasedAt,
    thumb,
    url,
    showCompleteDetails
}: InferProps<typeof Game.propTypes>) {
    return (
        <div>
            { showCompleteDetails &&
                <>
                    <h2>{title}</h2>
                    <h3>{releasedAt} - {platform}</h3>
                </>
            }
            <a href={url} target="_blank" rel="noreferrer">
                <img src={`/img/${thumb}`} alt={title} data-testid="game-screenshot"/>
            </a>
        </div>
    );
};

Game.propTypes = {
    title: PropTypes.string.isRequired,
    platform: PropTypes.string.isRequired,
    releasedAt: PropTypes.string.isRequired,
    thumb: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    showCompleteDetails: PropTypes.bool,
};

Game.defaultProps = {
    showCompleteDetails: false
};