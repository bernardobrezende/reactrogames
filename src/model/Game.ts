export interface Game {
    title: string;
    platform: string;
    releasedAt: string;
    thumb: string;
    url: string;
};
