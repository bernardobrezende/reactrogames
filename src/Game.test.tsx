import { render } from '@testing-library/react';
import Game from './Game';

describe('Game', () => {
    it('renders game information', () => {
        const game = {
            title: 'Golden Axe',
            url: 'https://www.retrogames.cz/play_037-SegaMS.php',
            platform: 'SMS',
            thumb: 'SMS-Golden_Axe.png'
        };
        const { asFragment } = render(<Game { ...game } />);
        expect(asFragment()).toMatchSnapshot();
    });
});