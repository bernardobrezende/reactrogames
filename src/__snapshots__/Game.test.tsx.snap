// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game initially renders only screenshot information 1`] = `
<DocumentFragment>
  <div>
    <a
      href="https://www.retrogames.cz/play_037-SegaMS.php"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Golden Axe"
        data-testid="game-screenshot"
        src="/img/SMS-Golden_Axe.png"
      />
    </a>
  </div>
</DocumentFragment>
`;

exports[`Game renders all game information 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Golden Axe
    </h2>
    <h3>
       - SMS
    </h3>
    <a
      href="https://www.retrogames.cz/play_037-SegaMS.php"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Golden Axe"
        data-testid="game-screenshot"
        src="/img/SMS-Golden_Axe.png"
      />
    </a>
  </div>
</DocumentFragment>
`;
